pkg/controller/cluster/cluster_controller.go

```diff
@@ -125,84 +145,112 @@ func (r *ReconcileCluster) Reconcile(request reconcile.Request) (reconcile.Resul
-			// TODO - Set to time.Duration depending on back-off behavior
-			//return reconcile.Result{RequeueAfter: time.Minute}, nil
-			return reconcile.Result{Requeue: true}, nil
+		if err != nil {
+			reqLogger.Error(err, "error updating cluster")
+			return reconcile.Result{}, err
 		}
 
-		// Don't requeue. We should get called to reconcile when the CR changes.
-		return reconcile.Result{}, nil
+		//get kubeconfig
+		var mode os.FileMode = 509
+		err = os.MkdirAll("./tmp", mode)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+
+		_, err = os.Create("tmp/config-" + kc.Name)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+
+		var config clusteroperatorv1alpha1.KubeConfig
+		config, err = k.GetKubeConfig(kc)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+
+		instance.Status.KubeConfig = config
+		reqLogger.Info("KUBECONFIG Updated")
+
+		//rolling udpates
+		os.Setenv("KUBECONFIG", "tmp/config-"+kc.Name)
+
+		//TODO: Right now, using defaults for intervals. Need to make changable
+		// Some changes will require rebuilding the nodes (for example, resizing nodes or changing the AMI)
+		// We call rolling-update to apply these changes
+		err = k.RollingUpdateCluster(kc)
+		if err != nil {
+			reqLogger.Error(err, "error performing rolling update on cluster")
+			return reconcile.Result{}, err
+		}
+		reqLogger.Info("Cluster Updated")
+		instance.Status.Phase = clusteroperatorv1alpha1.ClusterSetup
+		if err := r.client.Status().Update(context.TODO(), instance); err != nil {
+			return reconcile.Result{}, err
+		}
+		// instance.Status.Phase = clusteroperatorv1alpha1.ClusterSetup
+		// case clusteroperatorv1alpha1.ClusterSetup:
+		reqLogger.Info("Phase: SETUP")
+
+		// Setenv required if not using default .kube/config,
+		// the --kubeconfig option does not currently work for kops validate (1.18.2-alpha2)
+		os.Setenv("KUBECONFIG", "tmp/config-"+kc.Name)
+		status, err := k.ValidateCluster(kc)
+
+		instance.Status.KopsStatus = clusteroperatorv1alpha1.KopsStatus{}
+		if err != nil {
+			reqLogger.Info("Cluster Not Ready")
+			// instance.Status.Phase = clusteroperatorv1alpha1.ClusterPending
+		} else if len(status.Nodes) > 0 {
+			instance.Status.KopsStatus.Nodes = status.Nodes
+			reqLogger.Info("Cluster Created")
+			instance.Status.Phase = clusteroperatorv1alpha1.ClusterDone
+			reqLogger.Info("Phase: DONE")
+			if err := r.client.Status().Update(context.TODO(), instance); err != nil {
+				return reconcile.Result{}, err
+			}
+			//requeues every ten minutes to make sure its synced if any manual changes were done
+			return reconcile.Result{RequeueAfter: time.Minute * 10}, nil
+		} else {
+			// FIXME - If we get this state try validate again!!!
+			reqLogger.Info("Validate Returned Unexpected Result")
+			// instance.Status.Phase = clusteroperatorv1alpha1.ClusterPending
+		}
+
+		//It did not finish validating, requeue in five minutes
+		return reconcile.Result{RequeueAfter: time.Minute * 5}, nil
```

Comment:
what happens if the pod dies while this is being re-queued?

Reply:
When I start running the operator, it seems like it enqueues my resource even if I did not commit any events while the operator was running. That makes me think it just enqueues all resources on startup, but maybe I am missing something